test/test_futures.py

```diff
@@ -92,6 +92,41 @@ def test_chained_then(self):
         for i in range(len(futs)):
             self.assertEqual(futs[i].wait(), torch.ones(2, 2) + i + 1)
 
+    def test_add_done_callback_simple(self):
+        callback_result = False
+
+        def callback():
+            nonlocal callback_result
+            callback_result = True
+
+        fut = Future[torch.Tensor]()
+        fut.add_done_callback(callback)
+
+        self.assertFalse(callback_result)
+        fut.set_result(torch.ones(2, 2))
+        self.assertEqual(fut.wait(), torch.ones(2, 2))
+        self.assertTrue(callback_result)
+
+    def test_add_done_callback_maintains_callback_order(self):
```

Comment:
Let's add a test to verify that `add_done_callback` and `then` can be used together and runs in order.

Reply:
agreed, done